python/social_bot/envs/grocery_ground.py

```diff
@@ -191,35 +195,47 @@ def run(self, agent, world):
         agent_sentence = yield
         done = False
         while not done:
-            agent_pos = self.task_specific_observation()[:2]
-            vel_reward = 20.0 * np.linalg.norm(
-                agent_pos - pre_agent_pos, axis=-1)
-            pre_agent_pos = agent_pos
             agent_height = np.array(agent.get_link_pose('iCub::head'))[0][2]
             done = agent_height < 0.7
             alive_reward = agent_height - 0.7
-            reward = min((vel_reward + alive_reward), 1.0)
+            joint_pos = []
+            for joint_name in self._joints:
+                joint_state = self._agent.get_joint_state(joint_name)
+                joint_pos.append(joint_state.get_positions())
+            joint_pos = np.array(joint_pos).flatten()
+            movement_cost = np.sum(np.abs(joint_pos)) / joint_pos.shape[0]
+            reward = 2.0 * alive_reward - 0.5 * movement_cost
             agent_sentence = yield TeacherAction(reward=reward, done=done)
-
-    def task_specific_observation(self):
+    
+    @staticmethod
```

Comment:
Because we need to use the average pos of some key point for agent icub to calculate the distance to goal, static-method "get_icub_extra_obs()" was added to  class ICubStandingTask. So that we can access additional infomation of icub in other tasks.